### PR TITLE
Deprecate solve tracer

### PR DESCRIPTION
**Pull request by jwallwork23**
Addresses most of the points in #253, as well as updating the tracer and sediment model docs.

If these changes look okay then I will summarise the API changes in this PR and #251, put them in a page on the wiki and then publicise on Slack, etc.

### CODE REVIEW

File: thetis/coupled_timeintegrator_2d.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(self, solver):
         print_output('Coupled time integrator: {:}'.format(self.__class__.__name__))
         if not self.options.tracer_only:
             print_output('  Shallow Water time integrator: {:}'.format(self.swe_integrator.__name__))
-        if self.options.solve_tracer:
+        if self.options.tracer != {}:
```
Comment (tkarna) [inline thetis/coupled_timeintegrator_2d.py:40]: I wonder if there's a more succinct way to do this check, it's repeated in a few places. Maybe in `Solver2d.create_equations` you could add a boolean `self.solve_tracers = self.options.tracer != {}` to have it easily available? (Caveat is that this bool would not exist before `create_equations` is called but it's probably fine if it's used only internally. An alternative is to add something to the options itself).

File: thetis/options.py [hunk 2]
```diff
@@ -472,6 +472,8 @@ class TracerFieldOptions(FrozenHasTraits):
         None, allow_none=True, help='Source term for the tracer equation')
     diffusivity = FiredrakeScalarExpression(
         None, allow_none=True, help='Diffusion coefficient for the tracer equation')
+    conservative = Bool(
+        False, help='Should the tracer equation be solved in conservative form?')
```
Comment (tkarna) [inline thetis/options.py:476]: maybe use a more descriptive name e.g. `use_conservative_form`, similar to what we use for other boolean options

**Review by tkarna — APPROVED** [review]
Thanks, this is nice. Only a couple minor suggestions

**Review by tkarna — APPROVED** [review]
Thanks, looks good to me!

**Review by stephankramer — APPROVED** [review]
Looks perfect!

**Comment by jwallwork23** [comment]
Thanks! Will write a wiki page shortly.

**Pull request merged**